### ISSUE TITLE
hex_to_hash doesn't work on >16 length hashes

### ISSUE THREAD

**Issue by RiTu1337** [JohannesBuchner/imagehash#20]
I'd like to use hex_to_hash on 64 length hashes, however it only works on the 16 length hashes.

I've tried to edit the code, but the hash wouldn't be accurate.

**Comment by JohannesBuchner** [comment]
This should work for 16x16 hashes:
```
def hex_to_hash(hexstr):
	"""
	Convert a stored hash (hex, as retrieved from str(Imagehash))
	back to a Imagehash object.
	"""
	l = []
	if len(hexstr) != 2*(16*16)/8:
		raise ValueError('The hex string has the wrong length')
	for i in range(16*16/8):
		h = hexstr[i*2:i*2+2]
		v = int("0x" + h, 16)
		l.append([v & 2**i > 0 for i in range(8)])
	return ImageHash(numpy.array(l).reshape((16,16)))
```

Don't get confused with the 8s. 2 chars of hex representation encodes 8 bits. So to get the length you have to compute the hash array size (e.g. 16*16), divide by 8 and multiply by 2.

**Comment by RiTu1337** [comment]
Perfect. This works, thank you.

I've generated a phash (8, 16, 32) and dhash (8, 16, 32) hashes of every image ever shared on reddit (35m images) and I want to use this data to detect duplicate/very closely similar images (reposts). It seems like phash16 will be the best for it.

With numpy it takes 1.5us per hash to calculate the hamming distance (15s for 10m images) using np.count_nonzero(hash1!=hash2) so it seems doable! I have modified your code to return just the numpy array now that I have 35m hexes of each type.

If this is interesting to you, then I can publish the code after I'm done or send you a SQL dump.

Thanks again!
